src/eva/vision/data/datasets/segmentation/consep.py

```diff
@@ -106,8 +107,13 @@ def validate(self) -> None:
         )
 
     @override
-    def filename(self, index: int) -> str:
-        return os.path.basename(self._file_paths[self._get_dataset_idx(index)])
+    def filename(self, index: int, segmented: bool = True) -> str:
```

Comment:
For WSI classification we share the same `filename` for all patch embeddings, for two reasons:
1. Prevent explosion of number of `.pt` files
2. Facilitates loading of the embeddings, as there is a 1-1 mapping between `.pt` and corresponding image files

Can't we use the same approach here, and store the coordinates to the manifest file using the `metadata_keys` argument of the `EmbeddingsWriter` class?

Reply:
yes good idea, let me try that

Reply:
done!